### PR TITLE
Change Scan location and SSL support

### PR DESCRIPTION
**Pull request by weissi1994**
This pull request includes a

- [x] Bug fix
- [x] New feature
- [ ] Translation

The following changes were made

- Added js Function in map.js to perform post requests
- Added py Function in app.py to set new location
- Added Error Handling for json Decode Error
- Added SSL Support

### CODE REVIEW

File: static/map.js [hunk 1]
```diff
@@ -93,6 +117,13 @@ function initMap() {
     map.addListener('maptypeid_changed', function(s) {
         localStorage['map_style'] = this.mapTypeId;
     });
+    
+    google.maps.event.addListener(map, 'dblclick', function(event) {
+        console.log(event.latLng);
```
Comment (MRokas) [inline static/map.js]: `console.log()` shouldn't exist in prod. Forgot to remove I suppose?

Comment (medghaim) [inline static/map.js]: 'tis a `develop` branch after all!

Comment (weissi1994) [inline static/map.js]: Yes, forgot to remove it, its gone now

**Comment by fettlaus** [comment]
I don't really get why you created `set_loc(self)` instead of using or improving `next_loc(self)`. As I can see it, they are only slightly different.

**Comment by weissi1994** [comment]
You are right, I just didnt realize this function already existed :/

File: static/map.js [hunk 2]
```diff
@@ -121,7 +121,7 @@ function initMap() {
     google.maps.event.addListener(map, 'dblclick', function(event) {
         var lat = event.latLng.lat();
         var lng = event.latLng.lng();
-        post('/next_loc', {lat: lat, lng: lng});
+        post('/next_loc', {'lat': lat, 'lon': lng});
```
Comment (MRokas) [inline static/map.js]: Was this really necessary? It also goes against most code-styles.

**Comment by Ryan-Myers** [comment]
+1 for SSL Support. I've got my server set up with a pinned SSL cert, so it's been a pain to workaround it for this app. Thanks!

File: static/map.js (hunk 2 above)
Comment (weissi1994) [inline static/map.js]: yes it was^^, i get confused with all the coding languages i use all day

**Comment by AHAAAAAAA** [comment]
Change location implemented in #1225. Can you separate out the SSL functionality and do a PR for that?

**Pull request closed without merging**